serde-support/src/main/java/io/micronaut/serde/support/serializers/CoreSerializers.java

```diff
@@ -83,63 +103,78 @@ public boolean isEmpty(CharSequence value) {
     /**
      * A serializer for maps.
      *
-     * @param <K>  The key type
-     * @param <V>  The value type
+     * @param <K> The key type
+     * @param <V> The value type
      * @return A bit decimal serializer
      */
     @Singleton
     protected <K, V> Serializer<Map<K, V>> mapSerializer() {
-        return new Serializer<Map<K, V>>() {
+        return new SpecificOnlySerializer<Map<K, V>>() {
```

Comment:
Perhaps shift this to class as it is getting complex for an anonymous inner class